Mathlib/Analysis/MeanInequalities.lean

```diff
@@ -190,6 +190,75 @@ theorem geom_mean_eq_arith_mean_weighted_of_constant (w z : ι → ℝ) (x : ℝ
     ∏ i ∈ s, z i ^ w i = ∑ i ∈ s, w i * z i := by
   rw [geom_mean_weighted_of_constant, arith_mean_weighted_of_constant] <;> assumption
 
+/- **AM-GM inequality - equality condition**: This theorem provides the equality condition for the
+*positive* weighted version of the AM-GM inequality for real-valued nonnegative functions. -/
+theorem geom_mean_eq_arith_mean_weighted_iff_aux_of_pos (w z : ι → ℝ) (hw : ∀ i ∈ s, 0 < w i)
+    (hw' : ∑ i ∈ s, w i = 1) (hz : ∀ i ∈ s, 0 ≤ z i) :
+    ∏ i ∈ s, z i ^ w i = ∑ i ∈ s, w i * z i ↔ ∀ j ∈ s, z j = ∑ i ∈ s, w i * z i := by
```

Comment:
In this theorem, `w` and `z` can be inferred from the other arguments – you can't prove that `∀ i ∈ s, 0 < w i` without knowing what `w` is – so the usual library convention would be to make them implicit. That is, replace the round brackets with curly ones, `{w z : ι → ℝ}`. (Same applies to the other theorems here too.)

On another matter: is this first theorem only useful as a stepping-stone to the theorems later on in the file (as the `aux` in the name suggests)? If it is no longer useful once those theorems are proved (i.e. if the later theorems trivially imply it) then you can mark it as `private`, so it doesn't appear in the reference manual and can't be used outside this file.

Reply:
Aha! The linter shouldn't complain about the apostrophe as long as there is a doc-string. In your case you clearly meant to make a doc-string (the text preceding the theorem); but doc-strings need to start with `/--` (two dashes), not `/-` (one dash) which is just an ordinary comment. So this needs to be fixed anyway; and it will have the side-effect of stopping the linter complaining. 

I had overlooked this until now, so it's lucky that you made the remark about linters.

Reply:
Changing the arguments to implicit is a routine task but might be a little tedious. If you don't want to take on this re-factoring job, then don't feel you have to.